flowmachine/flowmachine/features/subscriber/contact_balance.py

```diff
@@ -69,59 +67,67 @@ def __init__(
         stop,
         *,
         hours="all",
-        table="all",
+        tables="all",
```

Comment:
Hmm, I'm in two minds about renaming this argument from `table` to `tables`. I think I've concluded that it's fine to change it (because we'll need to do cleanup on this soon anyway), but below are a few thoughts just for the general discussion & awareness.

In principle I agree that `tables` is a better name (although we also still allow a single string, in which case using the plural in `tables="<some_table_name>"` is also slightly confusing).

I'm not super happy with changing the name just in some clcasses because it introduces inconsistencies across the codebase that we should aim to eliminate. Before this PR almost every other class calls it `table`, while afterwards there is roughly a 50-50 mix of `table` and `tables`.

But as I said, I'm fine with it for now because the real underlying problems are that (i) we shouldn't really need to specify this in every instantiation of every class (I think?), and also (ii) this is one of the many examples (with other being aggregation units, subsets, etc.) where [primitive obsession](http://wiki.c2.com/?PrimitiveObsession) makes our lives hard and duplicates logic and arguments all over the codebase because we're using a string/list/... to capture a concept that needs richer logic and should really be captured in a separate class that can deal with that logic in a single place.

Anyway, long rambling but the bottom line is: no need to change it in this PR but let's ensure that subsequent refactorings ensure this is consistent again across the codebase and we have a more cohesive way of dealing with the underlying logic (the latter is mostly a note to self).

Reply:
However, if you're renaming the argument then it needs to be `tables` in the docstring above, too. (I can't seem to add a comment to that line directly?)

Reply:
I agree with all you mentioned above. A common pattern I have observed when creating new covariates is to basically copy the  `__init__` method from other covariates over and over again with small modifications between them (eg. some methods allow direction, others not). This creates inconsistencies and is rather tedious.  Also a lot of unit testing has to do with testing different combinations of parameters which are passed to `EventTablesUnion` the same way, which is a bit time consuming and redundant.

Reply:
I changed the docstring to `tables`, btw.